cmd/nfs-subdir-external-provisioner/provisioner.go

```diff
@@ -109,24 +149,39 @@ func (p *nfsProvisioner) Delete(volume *v1.PersistentVolume) error {
 	if err != nil {
 		return err
 	}
+
+	// Determine if the "onDelete" parameter exists.
+	// If it exists and has a delete value, delete the directory.
+	// If it exists and has a retain value, safe the directory.
+	onDelete := storageClass.Parameters["onDelete"]
+	switch onDelete {
+
+	case "delete":
+		return os.RemoveAll(oldPath)
+
+	case "retain":
+		return nil
+	}
+
 	// Determine if the "archiveOnDelete" parameter exists.
 	// If it exists and has a false value, delete the directory.
 	// Otherwise, archive it.
 	archiveOnDelete, exists := storageClass.Parameters["archiveOnDelete"]
 	if exists {
-		archiveBool, err := strconv.ParseBool(archiveOnDelete)
-		if err != nil {
-			return err
-		}
-		if !archiveBool {
-			return os.RemoveAll(oldPath)
+		if exists {
```

Comment:
why is this check repeating - 171 and 172?

Reply:
removed